components/tooltip/README.md

```diff
@@ -0,0 +1,66 @@
+<d2l-tooltip for="tooltip-error" state="error" align="start" offset="10">
+	Your error message will display here
+</d2l-tooltip>
+```
+
+### Boundaries
+
+Custom boundaries should be used sparingly. If no boundaries are provided, the viewport will be used. However, it might be necessary to constrain a tooltip to improve user experience or to prevent it from being cut off if it has an ancestor with `overflow: hidden;`.
+
+This can be done with the `boundary` attribute that allows any of the tooltip's `"top"`, `"bottom"`, `"left"` and `"right"` to be constrained. Boundaries are defined relative on the tooltip's [offset parent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent) so a boundary of `{"top": 0, "bottom": 0, "left": 0, "right": tooltip.offsetParent.width}` will constrain the tooltip so that it opens within its parent element's bounds. Note, `boundary.right` is defined relative to the offset parent's left side.
+
+In the following example the tooltip's orange offset parent has a width of `450px`. To constrain the tooltip to the dashed boundary we can set the top boundary to `50`, the bottom boundary to `10`, the left boundary to `100`, and the right boundary to `450`.
+
+![screenshot of a tooltip with custom boundaries](./screenshots/tooltip-boundary.png)
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/tooltip/tooltip.js';
+</script>
+
+<div class="offset-parent">
+	<button id="tooltip-boundary">Tooltip boundary</button>
+	<d2l-tooltip for="tooltip-boundary"
+		boundary="{&quot;top&quot;:50, &quot;bottom&quot;:10, &quot;left&quot;:100, &quot;right&quot;:450}">
+		This tooltip will not expand beyond its boundaries unless it is impossible to fit it inside
+	</d2l-tooltip>
+</div>
+```
+
+
+**Properties:**
+* `for` (required, String): provide the `id` of the tooltip's target element. If this attribute is not provided, the tooltip's parent element will be used as its target.
+* `align` (String): optionally align the tooltip with either the start or end of its target. If not set, the dropdown will attempt be centred. Valid values are: `start` and `end`.
+* `boundary` (Object) - optionally provide boundaries to where the tooltip will appear. Valid properties are `"top"`, `"bottom"`, `"left"` and `"right"`. The boundary is relative to the tooltip's [offset parent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent).
+* **DEPRECATED** `custom-target` (Object) - optionally provide a rectangle that the tooltip should position itself around. The object must provide values for `"left"`, `"top"`, `"width"`, and `"height"`. The custom target is relative to the tooltip's [offset parent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent). If a custom target is provided the user is responsible for showing and hiding the tooltip since it no longer has a target element that can be focused or hovered.
```

Comment:
Maybe we don't even mention it?

Reply:
The main reason I thought it might be worth mentioning is that way people don't discover it themselves and assume its good to use. It was never documented in the original `d2l-tooltip` but it still go used. That being said, I can definitely leave it out. I don't feel strongly either way.

Reply:
Gotcha. Is this one of the things you'd like to clean up and remove afterwards? If so, then we could likely safely exclude it. Otherwise it seems OK to leave the docs but maybe with a harsher "don't use this" vibe.

Reply:
I think I should be able to remove it entirely once I go in and clean up the places that use it. I'll exclude it and re-evaluate if something unexpected happens and I can't remove it entirely.